sympy/sets/sets.py

```diff
@@ -729,9 +729,10 @@ def __new__(cls, start, end, left_open=False, right_open=False):
             raise ValueError("Non-real intervals are not supported")
 
         # evaluate if possible
+        is_empty = (end - start).is_negative
         if (end < start) == True:
             return S.EmptySet
-        elif (end - start).is_negative:
+        elif is_empty:
```

Comment:
@asmeurer You are right! If I delete this `elif`, all test() will return True. So, I can delete that redundant